orderbook/src/bad_token/cache.rs

```diff
@@ -0,0 +1,67 @@
+use super::{BadTokenDetecting, TokenQuality};
+use anyhow::Result;
+use primitive_types::H160;
+use std::{collections::HashMap, sync::Mutex};
+
+pub struct CachingDetector {
+    inner: Box<dyn BadTokenDetecting>,
+    // std mutex is fine because we don't hold lock across await.
+    cache: Mutex<HashMap<H160, TokenQuality>>,
+}
+
+#[async_trait::async_trait]
+impl BadTokenDetecting for CachingDetector {
+    async fn detect(&self, token: H160) -> Result<TokenQuality> {
+        if let Some(result) = self.get_from_cache(&token) {
```

Comment:
Would it make sense to have a ttl for cache entries? I imaging some tokens can change their state (e.g. getting paused or a fee becoming enabled) so maybe caching results for 7 days on prod before we restart is a bit long?

Reply:
Good idea. Will remember this in an issue.